controllers/bucket_controller_test.go

```diff
@@ -977,6 +978,34 @@ func TestBucketReconciler_reconcileArtifact(t *testing.T) {
 				*conditions.TrueCondition(sourcev1.StorageOperationFailedCondition, sourcev1.InvalidPathReason, "is not a directory"),
 			},
 		},
+		{
+			name: "Ignore checksum should be set when spec ignore is in use",
+			beforeFunc: func(t *WithT, obj *sourcev1.Bucket, index *etagIndex, dir string) {
+				obj.Spec.Interval = metav1.Duration{Duration: interval}
+				obj.Spec.Ignore = pointer.StringPtr("!**.txt\n")
+			},
+			afterFunc: func(t *WithT, obj *sourcev1.Bucket, dir string) {
+				t.Expect(obj.Status.IgnoreChecksum).To(Equal("sha256:539530fbd29c252e357e36ddc065dc7923265e4ea795f09cada03287288f5e7a"))
+			},
+			want: sreconcile.ResultSuccess,
+			assertConditions: []metav1.Condition{
+				*conditions.TrueCondition(sourcev1.ArtifactInStorageCondition, meta.SucceededReason, "stored artifact for revision 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'"),
+			},
+		},
+		{
+			name: "Ignore checksum should be updated when spec ignore changes",
+			beforeFunc: func(t *WithT, obj *sourcev1.Bucket, index *etagIndex, dir string) {
+				obj.Spec.Interval = metav1.Duration{Duration: interval}
+				obj.Status.IgnoreChecksum = "sha256:539530fbd29c252e357e36ddc065dc7923265e4ea795f09cada03287288f5e7a"
+			},
+			afterFunc: func(t *WithT, obj *sourcev1.Bucket, dir string) {
+				t.Expect(obj.Status.IgnoreChecksum).To(BeEmpty())
+			},
+			want: sreconcile.ResultSuccess,
+			assertConditions: []metav1.Condition{
+				*conditions.TrueCondition(sourcev1.ArtifactInStorageCondition, meta.SucceededReason, "stored artifact for revision 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'"),
+			},
+		},
```

Comment:
Nice tests.
In the above, we test for cases where ignore is added and removed. Let's also test for the case where the existing value of ignore is updated, would result in a different checksum, not empty.

Reply:
Done!